File: service-1-java/src/main/java/com/nodalx/service1/MessageSender.java
package com.nodalx.service1;

import org.springframework.amqp.core.AmqpTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import javax.annotation.PostConstruct;

@Component
public class MessageSender {

    @Autowired
    private AmqpTemplate rabbitTemplate;

    @PostConstruct
    public void init() {
        sendPing();
    }

    public void sendPing() {
        String message = "ping from s1";
        System.out.println("Sending to s2: " + message);
        rabbitTemplate.convertAndSend("ping-pong-exchange", "service2.ping.key", message);
    }

    public void sendPong() {
        String message = "pong from s1";
        System.out.println("Sending to s2: " + message);
        rabbitTemplate.convertAndSend("ping-pong-exchange", "service2.pong.key", message);
    }
}

File: service-2-kotlin/src/main/kotlin/com/nodalx/service2/MessageSender.kt
package com.nodalx.service2

import org.springframework.amqp.core.AmqpTemplate
import org.springframework.stereotype.Component

@Component
class MessageSender(private val rabbitTemplate: AmqpTemplate) {

    fun sendPing() {
        val message = "Ping from s2"
        println("Sending to s1: $message")
        rabbitTemplate.convertAndSend("ping-pong-exchange", "service1.ping.key", message)
    }

    fun sendPong() {
        val message = "Pong from s2"
        println("Sending to s1: $message")
        rabbitTemplate.convertAndSend("ping-pong-exchange", "service1.pong.key", message)
    }
} 